### PR TITLE
Training speedups

### PR DESCRIPTION
**Pull request by gabrieltseng**
Three changes which speed up v1.2 model training by ~43 hrs.

| config | to 500k steps | ratio | median step | full run |
| --- | --- | --- | --- | --- | 
| unfused FSDP (v1.2, original) | 108.9 hrs | 1.000 | 0.636 s | 147 h (projected) | 
| + fused AdamW | 92.3 h | 0.848 | 0.534 s | ~130 h (projected) |
| + DDP | 80.4 h | 0.739 | 0.479 s | 107.2 h (completed) | 
| + proj-only target | 78.4 h | 0.720 | 0.465 s | 104.4 h (completed) |

Confirmed in https://wandb.ai/eai-ai2/2026_07_08_fused_adamw_e2e that the losses and the downstream evals stay consistent.

### CODE REVIEW

File: scripts/official/v1_2/base.py [hunk 1]
```diff
@@ -181,7 +181,9 @@ def build_train_module_config(
 ) -> ContrastiveLatentMIMTrainModuleConfig:
     """Build the train module config for an experiment."""
     return ContrastiveLatentMIMTrainModuleConfig(
-        optim_config=AdamWConfig(lr=0.0001, weight_decay=0.02, fused=False),
+        # fused=True: single-kernel AdamW update, ~20% faster steps at matched
+        # loss (validated vs run q3x9lvww in 2026_07_09_compile_fused_ab).
+        optim_config=AdamWConfig(lr=0.0001, weight_decay=0.02, fused=True),
```
Comment (favyen2) [inline scripts/official/v1_2/base.py:186]: Might be nice to have the speedup config here so it's obvious when branching off of it. (Like the ddp and lightweight target encoder parts.) Or comment here saying that the other version works the same but is faster.

File: olmoearth_pretrain/train/train_module/train_module.py [hunk 2]
```diff
@@ -443,6 +460,39 @@ def optim_step(self) -> None:
...
 
+    def _broadcast_params(self) -> None:
+        """Broadcast params and buffers from rank 0 so replicas start identical."""
+        if not (dist.is_available() and dist.is_initialized()):
+            return
+        group = self.dp_process_group
+        for t in itertools.chain(self.model.parameters(), self.model.buffers()):
+            dist.broadcast(t.detach(), src=0, group=group)
+
+    def _all_reduce_grads(self) -> None:
```
Comment (favyen2) [inline olmoearth_pretrain/train/train_module/train_module.py:471]: So when DDP is enabled, it handles the gradient syncing manually? Could finding a workaround where DDP built-in sync functionality can still be used make it faster? My AI agent used a similar all reduce approach for gradients for NAIP discriminator but it was saying the performance impact would be negligible since it was a very small part of the model, so I was thinking maybe there's additional speedup if you use the DDP approach where maybe it collects the gradients more efficiently (in parallel with the backward pass I think).

**Review by favyen2 — APPROVED** [review]
Looks great, thanks for finding these optimizations!

File: olmoearth_pretrain/train/train_module/train_module.py (hunk 2 above)
Comment (favyen2) [reply olmoearth_pretrain/train/train_module/train_module.py:471]: e.g. maybe we can make it run the two masked views together in one forward pass where they're concatenated on the batch dimension; probably requires some work though to thread it through the latent mim nn.Module.

Comment (gabrieltseng) [reply olmoearth_pretrain/train/train_module/train_module.py:471]: or get rid of the two forward passes

Comment (gabrieltseng) [reply olmoearth_pretrain/train/train_module/train_module.py:471]: i agree this is worth investigating but maybe out of scope for this PR? (or -- we should get this merged in to benefit from the speedups while we keep investigating)

File: scripts/official/v1_2/base.py (hunk 1 above)
Comment (gabrieltseng) [reply scripts/official/v1_2/base.py:186]: Added to `base_faster.py`
